src/tools/collectors.rs

```diff
@@ -0,0 +1,144 @@
+use std::collections::BTreeSet;
+
+use super::prelude::*;
+
+/// Collect data about blob
+#[derive(Debug)]
+pub struct BlobSummaryCollector {
+    header: BlobHeader,
+    records: usize,
+    keys: BTreeSet<Vec<u8>>,
```

Comment:
Now we have records with `deleted` flag, so we should count the number of such records and also should store all unique deleted keys